From 73fcf0b54f4c1fc07a2aa003dddaeb5a97177cce Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Thu, 31 Mar 2022 08:44:31 +0800
Subject: [PATCH 1/1] fix undefined bswap32 and bswap64 errors for MIPS machine

Replace bswap32 and bswap64 with internal defined version (__bswap_X)

Upstream Status: Submitted [libimobiledevice-devel@libimobiledevice.org]

Signed-off-by: Potin Lai <potin.lai@quantatw.com>
---
 include/endianness.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/endianness.h b/include/endianness.h
index 099877a..88b63db 100644
--- a/include/endianness.h
+++ b/include/endianness.h
@@ -113,8 +113,8 @@
      && !defined(__FLOAT_WORD_ORDER__)) \
  || (defined(__FLOAT_WORD_ORDER__) \
      && __FLOAT_WORD_ORDER__ == __ORDER_BIG_ENDIAN__)
-#define float_bswap64(x) bswap64(x)
-#define float_bswap32(x) bswap32(x)
+#define float_bswap64(x) __bswap_64(x)
+#define float_bswap32(x) __bswap_32(x)
 #else
 #define float_bswap64(x) (x)
 #define float_bswap32(x) (x)
-- 
2.17.1

